core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -355,4 +329,38 @@ private static String featureName(final URL url) {
         // fallback
         return Math.abs(url.hashCode()) + Features.EXTENSION;
     }
+
+    private static boolean isCukeSpaceTest(final TestClass testClass, final ClassLoader loader) {
+        Class<? extends Annotation> runWithType = null;
+        try {
+            runWithType = (Class<? extends Annotation>) loader.loadClass("org.junit.runner.RunWith");
+
+        } catch (final ClassNotFoundException error) {
+            // no-op
+        } catch (final NoClassDefFoundError e) {
+            // no-op
+        }
+
+        if (runWithType == null) {
+            return false;
+        }
+        final Annotation runWith = testClass.getAnnotation(runWithType);
+        if (runWith == null) {
+            return false;
+        }
+
+        try {
+            final Class<?> runner = Class.class.cast(runWithType.getMethod("value").invoke(runWith));
+            if ((!ArquillianCucumber.class.equals(runner) && !CukeSpace.class.equals(runner))) {

```

Comment:
I totally agree that the property must be decoupled from the implementation.
I was thinking at "archive processing" as a phase instead of linked to the name of the class.
Maybe I just wrongly used ArchiveProcessor instead of ArchiveProcessing.
So what about "skipArchiveProcessing" ?
The reason for that is because the "enrichment" is just a phase of the archive processing.
Does it make sense ?

About the Helper, I agree with you, anyway maybe for now should be enough to have just an helper class. Is it ok ?